### PR TITLE
Peer provides the necessary services for syncing

### PR DESCRIPTION
**Pull request by viquezclaudio**
Make sure peers provide the requires services for the different stages of syncing

## Pull request checklist

- [ ] All tests pass. The project builds and runs.
- [ ] I have resolved any merge conflicts.
- [ ] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: consensus/src/sync/syncer.rs [hunk 1]
```diff
@@ -215,6 +218,12 @@ impl<N: Network, M: MacroSync<N::PeerId>, L: LiveSync<N>> Syncer<N, M, L> {
         let blockchain = self.blockchain.clone();
         let network = Arc::clone(&self.network);
 
+        // Light nodes do not provide accounts chunks, so they are classified as incompatible.
```
Comment (viquezclaudio) [inline consensus/src/sync/syncer.rs:221]: Maybe move this to the "move_peer_into_live_sync" function, so that light nodes are not moved into live sync?

**Comment by nibhar** [comment]
I am not entirely familiar with this code so maybe this is an oversight on my end.

What about full nodes which are flagged incompatible by Full nodes macro sync as they don't provide History. Once the node transitions into live sync they are only required to provide AccountsChunks and FullBlocks which full nodes do provide. They need to be able to get out of the incompatible peer set now, as they are compatible with live sync.

**Review by Eligioo — APPROVED** [review]

**Comment by viquezclaudio** [comment]
This PR might no longer be necessary in favor of re-using the history store for full nodes according to #2692

**Comment by viquezclaudio** [comment]
Closing in favor of #2692

**Pull request closed without merging**